### ISSUE TITLE
linphone crash when app wake up from background by FCM push

### ISSUE THREAD

**Issue by djkskqyr3** [BelledonneCommunications/linphone-android#919]
I built linphone sdk from source code and I got library.
4.4.0-alpha.23+bbb41bfb
When app is in background, if app got incoming call by FCM push then I am getting crash error some times.
Here is build option and I used NDK android-ndk-r20b
cmake . -DLINPHONESDK_PLATFORM=Android -DENABLE_GPL_THIRD_PARTIES=NO -DENABLE_NON_FREE_CODECS=ON -DENABLE_VCARD=NO -DENABLE_ADVANCED_IM=NO -DENABLE_DB_STORAGE=NO -DENABLE_G729=ON -DENABLE_ASSETS=NO -DENABLE_QRCODE=NO
I have attached the crash log.
Please help me ASAP, Thank you

[crash_log.txt](https://github.com/BelledonneCommunications/linphone-android/files/4217359/crash_log.txt)

**Comment by Viish** [comment]
Hi,
Can you enable debug logs from the library and attach them please?
Also your linphonerc file would be helpful.
Thanks.

**Comment by djkskqyr3** [comment]
Thank you for your reply.
Here is linphonerc files.

And I will attach debug log.

[raw.zip](https://github.com/BelledonneCommunications/linphone-android/files/4218752/raw.zip)

**Comment by Viish** [comment]
Hi,
You are using a old and not stable version of the SDK.
Can you either try the 4.3.1 release version of the SDK or the latest 4.4.0-alpha ?
Thanks.

**Comment by djkskqyr3** [comment]
Please see my first message
I use 4.4.0-alpha.23+bbb41bfb

**Comment by Viish** [comment]
Not according to your logs:
```
2020-02-18 19:18:41.662 8905-8905/com.yabb I/Yabb: SDK VERSION=4.3.0-beta
2020-02-18 19:18:41.662 8905-8905/com.yabb I/Yabb: SDK BRANCH=tags/4.3.0-beta^0
```

**Comment by djkskqyr3** [comment]
I see, I will check again.
Wired, When I made linphone library from source code, it's named by 4.4.0-alpha.23+bbb41bfb

However, Thank you for your help,
I will let you know soon.

**Comment by djkskqyr3** [comment]
==== Phone information dump ====
DISPLAY NAME=PRA-TL10
DEVICE=HWPRA-H
MODEL=PRA-TL10
MANUFACTURER=HUAWEI
ANDROID SDK=26
ABIs=arm64-v8a, armeabi-v7a, armeabi
==== Linphone information dump ====
VERSION NAME=4.3.1+b273def
VERSION CODE=4200
PACKAGE=org.linphone.core
BUILD TYPE=release
SDK VERSION=4.3.1+b273def
SDK BRANCH=remotes/origin/release/4.3

I have a question, is it support x86 ?

**Comment by Viish** [comment]
Our pre-built binaries on maven don't include either x86 and x86_64 as there isn't much real devices using them, the sole purpose being the simulator.
If you want x86 you have to build the SDK yourself.
Also I just commited a fix in linphone-sdk on master branch that may solve your issue: 5b357240

**Comment by djkskqyr3** [comment]
However, there is another crash issues.

Crash1: Once device has been rebooted.
Crash2: Once package has been updated

![image](https://user-images.githubusercontent.com/1333215/74845981-b9f98e00-536a-11ea-9ddc-2359436a12a8.png)

**Comment by djkskqyr3** [comment]
Hi
I have tested with the latest master branch.
It's same result, it's crash.
I have attached the crash log.
Thank you

[full_crash_log1.txt](https://github.com/BelledonneCommunications/linphone-android/files/4227590/full_crash_log1.txt)

**Comment by Viish** [comment]
Yes but it is a different crash.
I fixed it in SDK commit 29b66021972f0521d3a1ae88391c50ef7b18abad, will be available on maven later today.

**Comment by djkskqyr3** [comment]
Thank you, 
How about the crash on BootReceiver?

**Comment by Viish** [comment]
Usually the crash `couldn't find "libc++_shared.so"` only happens when you install a new version of the app on the device and is fixed by clearing and rebuilding the project on Android Studio.
Maybe it has to do with the simulator, I never saw this issue on a real device only in the BootReceiver class.

**Comment by djkskqyr3** [comment]
It's crash on BootReceiver from real device. not Simulator
And there are many fabric crash logs from many users.
Also, normally I am rebuilding the project.

**Comment by djkskqyr3** [comment]
Sorry, still there is crash.

 ==== Phone information dump ====
 DISPLAY NAME=AOSP on IA Emulator
 DEVICE=generic_x86_arm
 MODEL=AOSP on IA Emulator
 MANUFACTURER=Google
 ANDROID SDK=28
 ABIs=x86, armeabi-v7a, armeabi
 ==== Linphone information dump ====
 VERSION NAME=4.4.0-alpha.94+471ecaa
 VERSION CODE=4200
 PACKAGE=org.linphone.core
 BUILD TYPE=release
 SDK VERSION=4.4.0-alpha.94+471ecaa


*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'google/sdk_gphone_x86_arm/generic_x86_arm:9/PSR1.180720.117/5875966:user/release-keys'
Revision: '0'
ABI: 'x86'
pid: 21170, tid: 21757, name: ServiceStart  >>> com.yabb <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x690076
    eax 00690076  ebx bad85ef4  ecx bca93138  edx bca93138
    edi bbd7d360  esi bc8da1d0
    ebp b716bfb8  esp b716bf8c  eip e58dfc2c
backtrace:
    #00 pc 00017c2c  /system/lib/libc.so (strcmp+12)
    #01 pc 002b5b75  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so (linphone_factory_find_supported_video_definition_by_name+68)
    #02 pc 002cbdb3  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so (linphone_core_set_preferred_video_definition_by_name+37)
    #03 pc 002c1c68  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so
    #04 pc 002c31e0  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so
    #05 pc 002c3334  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so (_linphone_core_new_with_config+121)
    #06 pc 002b56b3  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so
    #07 pc 002b5732  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so (linphone_factory_create_core_3+41)
    #08 pc 00326b33  /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/lib/x86/liblinphone.so (Java_org_linphone_core_FactoryImpl_createCore3+137)
    #09 pc 005f6a17  /system/lib/libart.so (art_quick_generic_jni_trampoline+71)
    #10 pc 005f0a02  /system/lib/libart.so (art_quick_invoke_stub+338)
    #11 pc 000a2f4e  /system/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+222)
    #12 pc 0029bb22  /system/lib/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+338)
    #13 pc 002950c6  /system/lib/libart.so (bool art::interpreter::DoCall<true, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+1014)
    #14 pc 005bf4fd  /system/lib/libart.so (MterpInvokeDirectRange+285)
    #15 pc 005e2fa1  /system/lib/libart.so (ExecuteMterpImpl+15265)
    #16 pc 000fe06a  /dev/ashmem/dalvik-classes4.dex extracted in memory from /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/base.apk!classes4.dex (deleted) (org.linphone.core.FactoryImpl.createCore+14)
    #17 pc 00266096  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.3007865774+598)
    #18 pc 0026c61c  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*, art::JValue*)+220)
    #19 pc 00293cab  /system/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+1019)
    #20 pc 005bc353  /system/lib/libart.so (MterpInvokeVirtual+691)
    #21 pc 005e2ba1  /system/lib/libart.so (ExecuteMterpImpl+14241)
    #22 pc 001c1400  /dev/ashmem/dalvik-classes2.dex extracted in memory from /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/base.apk!classes2.dex (deleted) (org.linphone.LinphoneManager.startLibLinphone+48)
    #23 pc 00266096  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.3007865774+598)
    #24 pc 0026c61c  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*, art::JValue*)+220)
    #25 pc 00293cab  /system/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+1019)
    #26 pc 005bc353  /system/lib/libart.so (MterpInvokeVirtual+691)
    #27 pc 005e2ba1  /system/lib/libart.so (ExecuteMterpImpl+14241)
    #28 pc 001bf76c  /dev/ashmem/dalvik-classes2.dex extracted in memory from /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/base.apk!classes2.dex (deleted) (org.linphone.LinphoneContext.start+44)
    #29 pc 00266096  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.3007865774+598)
    #30 pc 0026c61c  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*, art::JValue*)+220)
    #31 pc 00293cab  /system/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+1019)
    #32 pc 005bc353  /system/lib/libart.so (MterpInvokeVirtual+691)
    #33 pc 005e2ba1  /system/lib/libart.so (ExecuteMterpImpl+14241)
    #34 pc 001c6b90  /dev/ashmem/dalvik-classes2.dex extracted in memory from /data/app/com.yabb-v4AM1oJlPob1LRY1yd39Qw==/base.apk!classes2.dex (deleted) (org.linphone.YabbVOIPService$ServiceHandler.handleMessage+72)
    #35 pc 00266096  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.3007865774+598)
    #36 pc 0026c50e  /system/lib/libart.so (art::interpreter::EnterInterpreterFromEntryPoint(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*)+126)
    #37 pc 005a93fd  /system/lib/libart.so (artQuickToInterpreterBridge+1277)
    #38 pc 005f6aed  /system/lib/libart.so (art_quick_to_interpreter_bridge+77)
    #39 pc 000bbce5  /dev/ashmem/dalvik-jit-code-cache (deleted) (android.os.Handler.dispatchMessage+149)
    #40 pc 00063921  /dev/ashmem/dalvik-jit-code-cache (deleted) (android.os.Looper.loop+1393)
    #41 pc 005f0bd2  /system/lib/libart.so (art_quick_invoke_static_stub+418)
    #42 pc 000a2f5f  /system/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+239)
    #43 pc 0029bb22  /system/lib/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+338)
    #44 pc 00293cc8  /system/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+1048)
    #45 pc 005bdb1f  /system/lib/libart.so (MterpInvokeStatic+207)
    #46 pc 005e2d21  /system/lib/libart.so (ExecuteMterpImpl+14625)
    #47 pc 00ae3564  /system/framework/boot-framework.vdex (android.os.HandlerThread.run+56)
    #48 pc 00266096  /system/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.3007865774+598)
    #49 pc 0026c50e  /system/lib/libart.so (art::interpreter::EnterInterpreterFromEntryPoint(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*)+126)
    #50 pc 005a93fd  /system/lib/libart.so (artQuickToInterpreterBridge+1277)
    #51 pc 005f6aed  /system/lib/libart.so (art_quick_to_interpreter_bridge+77)
    #52 pc 005f0a02  /system/lib/libart.so (art_quick_invoke_stub+338)
    #53 pc 000a2f4e  /system/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+222)
    #54 pc 004d3209  /system/lib/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+89)
    #55 pc 004d44b7  /system/lib/libart.so (art::InvokeVirtualOrInterfaceWithJValues(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, jvalue*)+471)
    #56 pc 0050944c  /system/lib/libart.so (art::Thread::CreateCallback(void*)+1484)
    #57 pc 0008f065  /system/lib/libc.so (__pthread_start(void*)+53)
    #58 pc 0002485b  /system/lib/libc.so (__start_thread+75)

**Comment by Viish** [comment]
You built it yourself or took it from maven ?

**Comment by djkskqyr3** [comment]
I built it by myself.

![image](https://user-images.githubusercontent.com/1333215/75024865-24c6d880-54d5-11ea-8f9e-359a2b3c36e1.png)

**Comment by Viish** [comment]
Did you update the submodules ?

**Comment by djkskqyr3** [comment]
I have just GIT PULL

**Comment by djkskqyr3** [comment]
should I do git submodule update ?

**Comment by Viish** [comment]
No cmake won't do anything related to submodules. You have to updated them manually with `git submodule update`.

**Comment by djkskqyr3** [comment]
Thank you, Let me try again.

**Comment by djkskqyr3** [comment]
It seems that crash fixed. but please see below the log.

==== Phone information dump ====
DISPLAY NAME=AOSP on IA Emulator
DEVICE=generic_x86_arm
MODEL=AOSP on IA Emulator
MANUFACTURER=Google
ANDROID SDK=28
ABIs=x86, armeabi-v7a, armeabi
==== Linphone information dump ====
VERSION NAME=4.4.0-alpha.96+4928c46
VERSION CODE=4200
PACKAGE=org.linphone.core
BUILD TYPE=release
SDK VERSION=4.4.0-alpha.96+4928c46
SDK BRANCH=master

linphone_proxy_config_is_server_config_changed : 0
Publish params have changed on proxy config [0xc6276380]
Invalidating friends maps for list [0xe2faa010]
Couldn't find a supported video definition for name [qvga]
Video definition 'qvga' is not supported

Couldn't find a supported video definition for name [qvga]
Video definition 0x0 is not supported
Couldn't find a supported video definition for name [qvga]
Video definition 0x0 is not supported

**Comment by djkskqyr3** [comment]
Hi, any update?

**Comment by Viish** [comment]
No, I haven't saw your update until now.
Indeed there is still a crash, I will fix it.
In the meantime you can configure linphone to use a supported video definition (depends on your device).

**Comment by Viish** [comment]
By the way, it seems your app using our SDK (https://play.google.com/store/apps/details?id=com.yabb&hl=en_US) isn't respecting the GPL license (I couldn't see any mention of it anywhere). Please either update your app & website to reflect it or buy a proprietary license from Belledonne Communications (sales@belledonne-communications.com). With the license comes 6 months support for any issues (like this one).

**Comment by djkskqyr3** [comment]
Thank you for your reply.
Also here is another crash (when call terminated)

==== Phone information dump ====
DISPLAY NAME=Galaxy A3 (2016)
DEVICE=a3xelte
MODEL=SM-A310F
MANUFACTURER=samsung
ANDROID SDK=24
ABIs=armeabi-v7a, armeabi
==== Linphone information dump ====
VERSION NAME=4.4.0-alpha.113+593ee35
VERSION CODE=4200
PACKAGE=org.linphone.core
BUILD TYPE=release
SDK VERSION=4.4.0-alpha.113+593ee35
SDK BRANCH=master
[Context] Ready

signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
    r0 00000000  r1 00000060  r2 00000000  r3 00000000
    r4 b8b9ea88  r5 baad3800  r6 b8b9ea80  r7 bf8a5758
    r8 00000000  r9 baa105a0  sl 00000000  fp 00000000
    ip 00000000  sp bf8a5718  lr bb3d6bef  pc bb3d6c06  cpsr 60030030
backtrace:
    #00 pc 0007fc06  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #01 pc 0007396f  /data/app/com.yabb-2/lib/arm/libmediastreamer.so (ms_filter_process+38)
    #02 pc 00075417  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #03 pc 00075437  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #04 pc 00075437  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #05 pc 00075437  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #06 pc 00075437  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #07 pc 00075437  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #08 pc 0007530b  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #09 pc 0007517f  /data/app/com.yabb-2/lib/arm/libmediastreamer.so
    #10 pc 00047f93  /system/lib/libc.so (_ZL15__pthread_startPv+22)
    #11 pc 0001a161  /system/lib/libc.so (__start_thread+6)

**Comment by Viish** [comment]
This one is almost useful.
Please use debug library instead of release and send back the stacktrace (with symbols).

**Comment by djkskqyr3** [comment]
Hi, Any update?
Also We are going to put License link on App about page

**Comment by Viish** [comment]
As I said,

> Please use debug library instead of release and send back the stacktrace (with symbols).

**Comment by djkskqyr3** [comment]
Hi, I think that I send you crash log before.

https://github.com/BelledonneCommunications/linphone-android/issues/919#issuecomment-589637191

Would you please check it?
Thank you

**Comment by Viish** [comment]
The stacktrace isn't symbolized.
Use the debug version of the library, reproduce the issue and use ndk-stack tool to print a symbolized trace.

**Comment by djkskqyr3** [comment]
Hi, I found one strange log.
Couldn't find a supported video definition for name [cif] 

Here is full log

Registering all soundcard handlers                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Native android sound support not tested on SDK [27], disabled.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
[OpenSLES] libOpenSLES correctly loaded, creating OpenSLES MS soundcard                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Could not get board platform.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
This device (Google/Android SDK built for x86/) says it has no built-in echo canceller.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
No information available for [Google/Android SDK built for x86/],                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Using android >= 8 sound device descriptor.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Sound device information for [Google/Android SDK built for x86/] is: builtin=[yes], delay=[0] ms                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Could not get board platform.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
This device (Google/Android SDK built for x86/) says it has no built-in echo canceller.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
No information available for [Google/Android SDK built for x86/],                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Using android >= 8 sound device descriptor.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Sound device information for [Google/Android SDK built for x86/] is: builtin=[yes], delay=[0] ms                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
[OpenSLES] Using 720 for buffer size value                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
[OpenSLES] Using 48000 for sample rate value                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
Card 'openSLES: android sound card' added with capabilities [capture, playback]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
SDK version [27] detected                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Could not get board platform.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
This device (Google/Android SDK built for x86/) says it has no built-in echo canceller.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
No information available for [Google/Android SDK built for x86/],                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Using android >= 8 sound device descriptor.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Sound device information for [Google/Android SDK built for x86/] is: builtin=[yes], delay=[0] ms                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Card 'ANDROID SND (deprecated): Android Sound card' added with capabilities [capture, playback]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Registering all webcam handlers                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
[Legacy Capture] Android >= 8.0 detected, disabling legacy capture filter                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Webcam StaticImage: Static picture added                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
MSAndroidDisplay registered.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
MSAndroidDisplay (OpenGL ES2) registered (id=80).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
ms_factory_init_voip() done                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Found Android plugin libmsaaudio.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Found Android plugin libmsandroidcamera2.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Found Android plugin libmswebrtc.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Loading ms plugins from list                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
Loading plugin libmsaaudio.so...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Fail to load plugin libmsaaudio.so : dlopen failed: cannot locate symbol "AAudioStreamBuilder_setInputPreset" referenced by "/data/app/com.yabb-WkAdgunGanTNgZgSLOieCQ==/lib/x86/libmsaaudio.so"...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Could not locate init routine  of plugin libmsaaudio.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Loading plugin libmsandroidcamera2.so...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
[Camera2 Capture] libmsandroidcamera2 plugin loaded                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
[Camera2 Capture] Detecting cameras                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
[Camera2 Capture] Camera 0 is facing back with angle 90, hardware level is limited                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Webcam BackFacingCamera prepended                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
[Camera2 Capture] Camera 1 is facing front with angle 90, hardware level is full                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Webcam FrontFacingCamera prepended                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Plugin loaded (libmsandroidcamera2.so)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
Loading plugin libmswebrtc.so...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
libmswebrtc 1.1.1 plugin loaded, iSAC codec version 3.6.0, iLBC codec version 1.1.1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Plugin loaded (libmswebrtc.so)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Couldn't load all plugins in list                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Core callbacks [0xa2d04e00] registered on core [0x85ad6a00]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Codec opus/48000 fmtp=[useinbandfec=1] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec SILK/16000 fmtp=[] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec speex/16000 fmtp=[vbr=on] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Codec speex/8000 fmtp=[vbr=on] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Codec PCMU/8000 fmtp=[] number=0, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec PCMA/8000 fmtp=[] number=8, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec red/1000 fmtp=[] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec t140/1000 fmtp=[] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Codec GSM/8000 fmtp=[] number=3, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
Codec G722/8000 fmtp=[] number=9, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Codec iLBC/8000 fmtp=[mode=30] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
Codec AMR/8000 fmtp=[octet-align=1] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Codec AMR-WB/16000 fmtp=[octet-align=1] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec G729/8000 fmtp=[annexb=yes] number=18, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
Codec mpeg4-generic/16000 fmtp=[config=F8EE2000; constantDuration=512; indexDeltaLength=3; indexLength=3; mode=AAC-hbr; profile-level-id=76; sizeLength=13; streamType=5] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec mpeg4-generic/22050 fmtp=[config=F8EE2000; constantDuration=512; indexDeltaLength=3; indexLength=3; mode=AAC-hbr; profile-level-id=76; sizeLength=13; streamType=5] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec mpeg4-generic/32000 fmtp=[config=F8E82000; constantDuration=512; indexDeltaLength=3; indexLength=3; mode=AAC-hbr; profile-level-id=76; sizeLength=13; streamType=5] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec mpeg4-generic/44100 fmtp=[config=F8E82000; constantDuration=512; indexDeltaLength=3; indexLength=3; mode=AAC-hbr; profile-level-id=76; sizeLength=13; streamType=5] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec mpeg4-generic/48000 fmtp=[config=F8EE2000; constantDuration=512; indexDeltaLength=3; indexLength=3; mode=AAC-hbr; profile-level-id=76; sizeLength=13; streamType=5] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec iSAC/16000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Codec speex/32000 fmtp=[vbr=on] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Codec SILK/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec SILK/12000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Codec SILK/24000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Codec G726-16/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Codec G726-24/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Codec G726-32/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Codec G726-40/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Codec AAL2-G726-16/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Codec AAL2-G726-24/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Codec AAL2-G726-32/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Codec AAL2-G726-40/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Codec CODEC2/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec BV16/8000 fmtp=[] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec VP8/90000 fmtp=[] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Codec H264/90000 fmtp=[profile-level-id=42801F] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Codec H265/90000 fmtp=[] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec MP4V-ES/90000 fmtp=[profile-level-id=3] number=-1, default enablement: 16) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
Codec H263-1998/90000 fmtp=[CIF=1;QCIF=1] number=-1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Codec H263/90000 fmtp=[] number=34, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Codec 1016/8000 fmtp=[] number=1, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Codec G723/8000 fmtp=[] number=4, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Codec LPC/8000 fmtp=[] number=7, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
Codec L16/44100 fmtp=[] number=10, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec L16/44100 fmtp=[] number=11, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Codec CN/8000 fmtp=[] number=13, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
Codec H261/90000 fmtp=[] number=31, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Codec MPV/90000 fmtp=[] number=32, default enablement: 0) added to the list of possible codecs.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Root ca path set to /system/etc/security/cacerts                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Root ca path set to /system/etc/security/cacerts                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
[Platform Helper] getDnsServers() returning 10.0.2.3,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
AndroidPlatformHelpers found DNS server 10.0.2.3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Found Android plugin libmsaaudio.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Found Android plugin libmsandroidcamera2.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Found Android plugin libmswebrtc.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Loading ms plugins from list                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
Loading plugin libmsaaudio.so...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Fail to load plugin libmsaaudio.so : dlopen failed: cannot locate symbol "AAudioStreamBuilder_setInputPreset" referenced by "/data/app/com.yabb-WkAdgunGanTNgZgSLOieCQ==/lib/x86/libmsaaudio.so"...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Could not locate init routine  of plugin libmsaaudio.so                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Loading plugin libmsandroidcamera2.so...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
[Camera2 Capture] libmsandroidcamera2 plugin loaded                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Plugin loaded (libmsandroidcamera2.so)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
Loading plugin libmswebrtc.so...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
libmswebrtc 1.1.1 plugin loaded, iSAC codec version 3.6.0, iLBC codec version 1.1.1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Plugin loaded (libmswebrtc.so)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Couldn't load all plugins in list                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
linphone_core_set_playback_gain_db(): no active call.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
linphone_core_set_mic_gain_db(): no active call.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
[ToneManager] create ToneManager()                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Starting stun server resolution [54.66.233.186]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Stun server resolution successful.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Sal nat helper [enabled]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Sal use rports [enabled]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
MTU is supposed to be 1300, rtp payload max size will be 1240                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
Could not find encoder for SILK                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for AMR                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Could not find encoder for AMR-WB                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Could not find encoder for mpeg4-generic                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
: uid=10080(com.yabb) identical 3 lines                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Could not find encoder for mpeg4-generic                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Could not find encoder for SILK                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
: uid=10080(com.yabb) identical 1 line                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
Could not find encoder for SILK                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for G726-16                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Could not find encoder for G726-24                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Could not find encoder for G726-32                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Could not find encoder for G726-40                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Could not find encoder for AAL2-G726-16                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Could not find encoder for AAL2-G726-24                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Could not find encoder for AAL2-G726-32                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Could not find encoder for AAL2-G726-40                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Could not find encoder for CODEC2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Could not find encoder for BV16                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for 1016                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for G723                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for LPC                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Could not find encoder for CN                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
Could not find encoder for H265                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for MP4V-ES                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Could not find encoder for H263-1998                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
Could not find encoder for H263                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for H261                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Could not find encoder for MPV                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Supported codec red/1000 fmtp= automatically added to codec list.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Supported codec t140/1000 fmtp= automatically added to codec list.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Sal use rports [enabled]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
belle_sip_stack_set_well_know_port() : set to [5060]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
belle_sip_stack_set_well_know_port_tls() : set to [5061]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Root ca path set to /data/user/0/com.yabb/files/share/linphone/rootca.pem                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Sal::unlistenPorts(): done                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
Creating listening point [0x8b978e80] on [sip:[::0]:-1;transport=UDP]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
Random UDP port is 47796                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Creating listening point [0x8b978fc0] on [sip:[::0]:-1;transport=TCP]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
Random TCP port is 48022                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Creating listening point [0x8acd7180] on [sip:[::0]:-1;transport=TLS]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
Random TCP port is 59623                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Root ca path set to /system/etc/security/cacerts                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Root ca path set to /system/etc/security/cacerts                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Root ca path set to /data/user/0/com.yabb/files/share/linphone/rootca.pem                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
linphone_proxy_config_is_server_config_changed : 0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Publish params have changed on proxy config [0xaabf11e0]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Invalidating friends maps for list [0xa2d6a110]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Couldn't find a supported video definition for name [cif]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Video definition 'cif' is not supported                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Listening point [0x8b978e80] on [sip:[::0]:47796;transport=UDP] destroyed                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Listening point [0x8b978fc0] on [sip:[::0]:48022;transport=TCP] destroyed                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Listening point [0x8acd7180] on [sip:[::0]:59623;transport=TLS] destroyed                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Sal::unlistenPorts(): done                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
Creating listening point [0x8acd7280] on [sip:0.0.0.0:-1;transport=UDP]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Random UDP port is 45335                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Creating listening point [0x8acd7380] on [sip:0.0.0.0:-1;transport=TCP]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Random TCP port is 41291                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Creating listening point [0x8b978fc0] on [sip:0.0.0.0:-1;transport=TLS]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Random TCP port is 54185                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Root ca path set to /system/etc/security/cacerts                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Root ca path set to /system/etc/security/cacerts                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
Root ca path set to /data/user/0/com.yabb/files/share/linphone/rootca.pem                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Core callbacks [0xa2d051c0] registered on core [0x85ad6a00]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Core callbacks [0xa2d039f0] registered on core [0x85ad6a00]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
[Manager] We are here because of a received push notification, enter background mode before starting the Core                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
[Context] Global state is [Startup]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Linphone core [0x85ad6a00] notified [global_state_changed]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
Root ca path set to /data/user/0/com.yabb/files/share/linphone/rootca.pem                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
[Platform Helper] onLinphoneCoreStart, network monitoring is true                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
[Platform Helper] Registering network callbacks                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
[Platform Helper] [Network Manager 26] ACCESS_NETWORK_STATE permission is granted                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
vice: requestNetwork for uid/pid:10080/17002 NetworkRequest [ TRACK_DEFAULT id=34, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VPN] ]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
quest NetworkRequest [ TRACK_DEFAULT id=34, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VPN] ] with score 60                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
st NetworkRequest [ TRACK_DEFAULT id=34, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VPN] ] with score 60                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
tworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=34, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VPN] ] with score 60                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
[Platform Helper] Registering doze receiver                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
[Platform Helper] Registering interactivity receiver                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
[Platform Helper] No connectivity: setting network unreachable                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
[Context] Global state is [Configuring]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
Linphone core [0x85ad6a00] notified [global_state_changed]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
[Context] Configuring state is [Skipped]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
Linphone core [0x85ad6a00] notified [configuring_status]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
[Context] Global state is [On]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
Linphone core [0x85ad6a00] notified [global_state_changed]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
[Manager] Configuring Core                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
othAdapter: Bluetooth binder is null                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
Core callbacks [0xa2d05790] registered on core [0x85ad6a00]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
Table already up to date: duplicate column name: call_id.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
linphone_core_get_call_history(): completed in 1 ms                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
nothing to migrate, skipping...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Couldn't find a supported video definition for name [cif]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Video definition 0x0 is not supported                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
Couldn't find a supported video definition for name [cif]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
Video definition 0x0 is not supported                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
[Manager] MediaStreamer : 2 cores detected and configured                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
[Manager] Found front facing camera: FrontFacingCamera                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
[Manager] Core configured                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
H264Helper setH264Mode  MODE_AUTO - Mode = Auto                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
H264Helper setH264Mode  MODE_AUTO 1 - Mode = Auto                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
H264Helper Openh264 disabled on the project, now using MediaCodec                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
Cannot enable/disable unknown filter [MSOpenH264Dec] on factory [0xa1667a60]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
Cannot enable/disable unknown filter [MSOpenH264Enc] on factory [0xa1667a60]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
Cannot enable/disable unknown filter [MSMediaCodecH264Dec] on factory [0xa1667a60]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Cannot enable/disable unknown filter [MSMediaCodecH264Enc] on factory [0xa1667a60]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
H264Helper setH264Mode - Mode = Auto                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
[Manager] Phone state is idle

**Comment by djkskqyr3** [comment]
I know that cif is default video definition for factory.

ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_CIF_W, MS_VIDEO_SIZE_CIF_H, "cif");

I am not sure when I can get this error?
**Couldn't find a supported video definition for name [cif]**

**Comment by Viish** [comment]
This simply means the hardware camera on your device doesn't support CIF format.

**Comment by djkskqyr3** [comment]
Ok, I agree with you, x86 simulator does not support cif probably.
Thank you for your help

Sent from my iPhone

On Apr 12, 2020, at 8:23 PM, Sylvain Berfini <notifications@github.com<mailto:notifications@github.com>> wrote:


This simply means the hardware camera on your device doesn't support CIF format.

**Comment by DroidPulkit** [comment]
> However, there is another crash issues.
> 
> Crash1: Once device has been rebooted.
> Crash2: Once package has been updated
> 
> The reason, Factory can not load libraries.
> Here is the crash log.
> 
> Fatal Exception: java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/system/framework/com.google.android.maps.jar", zip file "/data/app/com.yabb-1/base.apk"],nativeLibraryDirectories=[/data/app/com.yabb-1/lib/arm, /vendor/lib, /system/lib]]] couldn't find "libc++_shared.so"
> at java.lang.Runtime.loadLibrary + 379(Runtime.java:379)
> at java.lang.System.loadLibrary + 988(System.java:988)
> at org.linphone.core.FactoryImpl. + 295(FactoryImpl.java:295)
> at org.linphone.core.Factory.instance + 44(Factory.java:44)
> at org.linphone.LinphonePreferences.getConfig + 172(LinphonePreferences.java:172)
> at org.linphone.LinphonePreferences.isAutoStartEnabled + 1028(LinphonePreferences.java:1028)
> at org.linphone.BootReceiver.onReceive + 41(BootReceiver.java:41)
> at android.app.ActivityThread.handleReceiver + 2614(ActivityThread.java:2614)
> at android.app.ActivityThread.access$1700 + 156(ActivityThread.java:156)
> at android.app.ActivityThread$H.handleMessage + 1385(ActivityThread.java:1385)
> at android.os.Handler.dispatchMessage + 102(Handler.java:102)
> at android.os.Looper.loop + 135(Looper.java:135)
> at android.app.ActivityThread.main + 5314(ActivityThread.java:5314)
> at java.lang.reflect.Method.invoke(Method.java)
> at java.lang.reflect.Method.invoke + 373(Method.java:373)
> at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 903(ZygoteInit.java:903)
> at com.android.internal.os.ZygoteInit.main + 698(ZygoteInit.java:698)
> 
> ![image](https://user-images.githubusercontent.com/1333215/74845981-b9f98e00-536a-11ea-9ddc-2359436a12a8.png)

Usually, you just need to replace the Log.i() to android.util.Log.i() as sometimes the linphone lib is not loaded, and Log.i() is function of the library

**Comment by djkskqyr3** [comment]
You are misunderstand, 
I am not getting the crash from Log.i()
I was getting the crash on 
LinphonePreferences.instance.isAutoStartEnabled();
LinphonePreferences.instance.getsServiceNotificationVisibility();

**Comment by djkskqyr3** [comment]
Hi @Viish good day
Still I have many crash on video definition.

Here is my config for video.

[net]
#Because dynamic bitrate adaption can increase bitrate, we must allow "no limit"
download_bw=0
upload_bw=0

[video]
**size=qvga**


static void initialize_supported_video_definitions(LinphoneFactory *factory) {
#if !defined(__ANDROID__) && !TARGET_OS_IPHONE
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_1080P_W, MS_VIDEO_SIZE_1080P_H, "1080p");
#endif
#if !defined(__ANDROID__) && !TARGET_OS_MAC /*limit to most common sizes because mac video API cannot list supported resolutions*/
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_UXGA_W, MS_VIDEO_SIZE_UXGA_H, "uxga");
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_SXGA_MINUS_W, MS_VIDEO_SIZE_SXGA_MINUS_H, "sxga-");
#endif
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_720P_W, MS_VIDEO_SIZE_720P_H, "720p");
#if !defined(__ANDROID__) && !TARGET_OS_MAC
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_XGA_W, MS_VIDEO_SIZE_XGA_H, "xga");
#endif
#if !defined(__ANDROID__) && !TARGET_OS_IPHONE
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_SVGA_W, MS_VIDEO_SIZE_SVGA_H, "svga");
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_4CIF_W, MS_VIDEO_SIZE_4CIF_H, "4cif");
#endif
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_VGA_W, MS_VIDEO_SIZE_VGA_H, "vga");
#if TARGET_OS_IPHONE
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_IOS_MEDIUM_H, MS_VIDEO_SIZE_IOS_MEDIUM_W, "ios-medium");
#endif
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_CIF_W, MS_VIDEO_SIZE_CIF_H, "cif");
#if !TARGET_OS_MAC || TARGET_OS_IPHONE /* OS_MAC is 1 for iPhone, but we need QVGA */
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_QVGA_W, MS_VIDEO_SIZE_QVGA_H, "qvga");
#endif
	ADD_SUPPORTED_VIDEO_DEFINITION(factory, MS_VIDEO_SIZE_QCIF_W, MS_VIDEO_SIZE_QCIF_H, "qcif");
}


Android does not support **qvga** ?
